eventmetrics/parsers/parserValidator.go

```diff
@@ -0,0 +1,191 @@
+package parsers
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/xmidt-org/interpreter"
+	"github.com/xmidt-org/interpreter/history"
+	"github.com/xmidt-org/interpreter/validation"
+	"go.uber.org/zap"
+)
+
+var (
+	errFatal      = errors.New("fatal error")
+	errValidation = errors.New("validation error")
+)
+
+// EventsParser parses the relevant events from a device's history of events and returns those events.
+type EventsParser interface {
+	Parse(eventsHistory []interpreter.Event, currentEvent interpreter.Event) ([]interpreter.Event, error)
+}
+
+type parserValidator struct {
+	cycleParser              EventsParser
+	cycleValidator           history.CycleValidator
+	eventsValidator          validation.Validator
+	shouldActivate           func([]interpreter.Event, interpreter.Event) bool
+	eventsValidationCallback func(interpreter.Event, bool, error)
+	cycleValidationCallback  func(bool, error)
+}
+
+func (p *parserValidator) Validate(events []interpreter.Event, currentEvent interpreter.Event) (bool, error) {
+	p.setDefaults()
```

Comment:
Can the parserValidator be called from multiple goroutines? This doesn't seem concurrency safe?  One option I know of is to use [`sync.Once`](https://pkg.go.dev/sync#Once.Do), since I think we only need to set defaults once at the beginning? But that still won't block other goroutines from continuing on while one goroutine is making sure there are no nils...Is it possible to make a `NewParserValidator()` function and avoid this entirely?